src/hdrgen.d

```diff
@@ -1746,7 +1746,12 @@ public:
         typeToBuffer(tf, f.ident);
         if (hgs.hdrgen == 1)
         {
-            if (!f.type.nextOf || f.storage_class & STCauto)
+            if(f.fbody.toChars()[0] == '\0')
+            {
```

Comment:
are you sure you want to write a declaration here ?
and you should not relay on toChars to do that check.